### ISSUE TITLE
Error: The script will never generate a response.

### ISSUE THREAD

**Issue by autodepo** [signalnerve/websocket-template#6]
Somehow, I always get `Error: The script will never generate a response.` when the client closes the connection.
It doesn't affect the functionality, but really spike up the numbers of errors detected.

Is there any workaround?

**Comment by madurangae** [comment]
I'm having the same issue.

**Comment by MachineHerald007** [comment]
> Somehow, I always get `Error: The script will never generate a response.` when the client closes the connection. It doesn't affect the functionality, but really spike up the numbers of errors detected.
> 
> Is there any workaround?

The issue is ultimately because this template sharing promises in the global scope. When **request a** awaits a promise and **request b** resolves it, **request a's** code continues to run regardless of if another request resolves it because Cloudflare assumes it's executing on the other request's behalf.

The only way to avoid this is to not have any request touch the global scope and utilize Cloudflare's caching-- like durable objects, for instance. Not putting promises in the global scope and using Cloudflare's durable objects for websockets, resolved this issue.

**Comment by MachineHerald007** [comment]
> Somehow, I always get `Error: The script will never generate a response.` when the client closes the connection. It doesn't affect the functionality, but really spike up the numbers of errors detected.
> 
> Is there any workaround?
Yes. You're going to have to use durable objects for your workers to handle websocket connections.

The issue is ultimately because this template is sharing promises in the global scope. When request a awaits a promise and request b resolves it, request a's code continues to run regardless of if another request resolves it because Cloudflare assumes it's executing on the other request's behalf.

The only way to avoid this is to not have any request touch the global scope and utilize Cloudflare's caching-- like durable objects, for instance. Not putting promises in the global scope and using Cloudflare's durable objects for websockets, resolved this issue.

**Comment by apuravchauhan** [comment]
Whats the solution to this?
Also seeing this error on close
✘ [ERROR] Uncaught (in response) Error: The Workers runtime canceled this request because it detected that your Worker's code had hung and would never generate a response.
